src/Samsung.java

```diff
@@ -0,0 +1,54 @@
+
+public abstract class Samsung implements Phone {
```

Comment:
what happens when we also have to deal with other phone manufacturers? we reimplement all functionality?